app/models/conference.rb

```diff
@@ -595,7 +614,20 @@ def notify_on_registration_dates_changed?
   end
 
   def registration_limit_exceeded?
-    registration_limit > 0 && registrations.count >= registration_limit
+    registration_limit > 0 && registrations.count + unregistered_speakers >= registration_limit
+  end
+
+  # Returns the number of speakers of new, unconfirmed and confirmed events that are not registered
+  def unregistered_speakers
+    pending_speakers = []
+    program.events.each do |event|
+      if %w(new unconfirmed confirmed).include? event.state
```

Comment:
This means we will restrict the number of registrations, based on submitted proposals that may or may not happen. I don't agree with that.

When implementing features, it is best that we first agree on what we are trying to achieve / how do we want things to work.